File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        int min_count = 0; 
        int open_deficit = 0;

        for (char c: s.toCharArray()) { 
            if (c == '(') {
                open_deficit += 1;
            } else {
                if (open_deficit == 0) {
                    min_count += 1;
                } else {
                    open_deficit -= 1;
                }
            }
        }
        return min_count + open_deficit;
    }
}

File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        int res = 0;
        int numOpen = 0;
        for (char c : s) {
            if (c == '(') {
                ++numOpen;
            } else {
                numOpen > 0 ? numOpen-- : res++;
            }
        }
        return numOpen + res;
    }
};